### ISSUE TITLE
What if the host does not have an email server but uses external services like Amazon?

### ISSUE THREAD

**Issue by Clanwarz** [amaurymartiny/check-if-email-exists#2]
I have tried your scripts but they fail when the host does not have their own email server.

**Comment by amaurymartiny** [comment]
I am aware of this problem. Do you have any idea how to solve this?

**Comment by Clanwarz** [comment]
Use the MX records instead of the hostname

```
root@0003 ~ # dig +short oracle.com mx | sed 's/\.$//g'
20 aserp2050.oracle.com
20 aserp2030.oracle.com
20 aserp2040.oracle.com
20 aserp2020.oracle.com
20 userp2020.oracle.com
20 userp2050.oracle.com
20 userp2040.oracle.com
20 userp2030.oracle.com
```

**Comment by Clanwarz** [comment]
Amaurymartiny, I see you updated the scripts. What do you think of the project now? What remains to be done that you are aware of?

**Comment by amaurymartiny** [comment]
Hey! I can only think of small improvements:
- create a binary so that it can be run on windows too
- in that binary, create threads so that we can test multiple domains/port at the same time, resulting in less time to find if an email exists

If you have any others, I would really want to hear them! Do create separate issues, it'll be easier to discuss.

**Comment by Clanwarz** [comment]
I haven't forgotten about this thread. I'll have some time later to check and reply.

**Comment by amaurymartiny** [comment]
Btw, I implemented my last comment in #7.
